es-vue-base/src/lib-components/EsNavBar.vue

```diff
@@ -300,28 +335,34 @@ export default {
         const overlay = document.querySelector('.content-overlay');
 
         // Search bar elements for hiding/showing
-        const searchBar = document.querySelector('.nav-search-bar');
+        const searchBarDesktop = document.querySelector('.nav-search-bar-desktop');
+        const searchBarMobile = document.querySelector('.nav-search-bar-mobile');
         const productMenu = document.querySelector('.product-menu');
-        const searchIcon = document.querySelector('.search-icon');
-        const searchForm = document.getElementById('searchBar');
+        const searchIconMobile = document.querySelector('.search-icon-mobile');
+        const searchIconDesktop = document.querySelector('.search-icon-desktop');
+        const searchForm = document.getElementById('searchBarDesktop');
 
         // Function to show/hide search bar
         function toggle_search_bar(show_search_bar) {
```

Comment:
Why not split this function into two, one for desktop and one for mobile? Then you would know which one to give focus to.

This function is already called from a specific click listener for the mobile search button, and a specific click listener for the desktop search button, so it should be straightforward to call the mobile or desktop versions of this function from those places, and call whatever the correct one is from all other places this function is called.

Reply:
That helped, thanks. The rest of the PR should be ready to review.

I was referring to one specific edge case when the user opens the mobile account menu then opens the mobile search menu. The menus overlap and it looks bad.

As far as I can tell, it would require pasting the contents of `collapse_mobile_menus()` and the functions it calls into `toggle_search_bar_mobile()`, which is defined first. It gets confusing.

I don't think this is critical enough to hold up the PR, but we should address it in the near future.

Reply:
Looks fixed to me! If I open the search bar and then open either the account menu or the nav menu, the search bar closes. I'm not able to open an account/nav menu and then open the search bar, since those take up the whole screen, so I think we're fine there.